src/main/java/org/breedinginsight/brapps/importer/model/imports/germplasm/GermplasmImportService.java

```diff
@@ -17,57 +17,38 @@
 
 package org.breedinginsight.brapps.importer.model.imports.germplasm;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import io.micronaut.http.HttpStatus;
-import io.micronaut.http.exceptions.HttpStatusException;
-import io.micronaut.http.server.exceptions.InternalServerException;
 import lombok.extern.slf4j.Slf4j;
-import org.brapi.client.v2.model.exceptions.ApiException;
-import org.brapi.v2.model.core.BrAPIProgram;
-import org.brapi.v2.model.germ.*;
-import org.breedinginsight.brapps.importer.daos.BrAPIGermplasmDAO;
-import org.breedinginsight.brapps.importer.daos.BrAPIProgramDAO;
-import org.breedinginsight.brapps.importer.daos.ImportDAO;
 import org.breedinginsight.brapps.importer.model.ImportUpload;
-import org.breedinginsight.brapps.importer.model.base.*;
 import org.breedinginsight.brapps.importer.model.imports.BrAPIImport;
 import org.breedinginsight.brapps.importer.model.imports.BrAPIImportService;
-import org.breedinginsight.brapps.importer.model.imports.PendingImport;
 import org.breedinginsight.brapps.importer.model.response.ImportPreviewResponse;
-import org.breedinginsight.brapps.importer.model.response.ImportPreviewStatistics;
-import org.breedinginsight.brapps.importer.model.response.PendingImportObject;
-import org.breedinginsight.brapps.importer.model.response.ImportObjectState;
+import org.breedinginsight.brapps.importer.services.processors.*;
 import org.breedinginsight.model.Program;
 import org.breedinginsight.services.exceptions.DoesNotExistException;
 import org.breedinginsight.services.exceptions.UnprocessableEntityException;
-import org.jooq.JSONB;
+import org.breedinginsight.services.exceptions.ValidatorException;
 import tech.tablesaw.api.Table;
 
 import javax.inject.Inject;
+import javax.inject.Provider;
 import javax.inject.Singleton;
 import java.util.*;
-import java.util.stream.Collectors;
 
 @Singleton
 @Slf4j
 public class GermplasmImportService extends BrAPIImportService {
 
     private String IMPORT_TYPE_ID = "GermplasmImport";
 
-    private BrAPIGermplasmDAO brAPIGermplasmDAO;
-    private BrAPIProgramDAO brAPIProgramDAO;
-    private ImportDAO importDAO;
-    private ObjectMapper objMapper;
+    private Provider<GermplasmProcessor> germplasmProcessorProvider;
+    private Provider<ProcessorManager> processorManagerProvider;
 
     @Inject
-    public GermplasmImportService(BrAPIProgramDAO brAPIProgramDAO, BrAPIGermplasmDAO brAPIGermplasmDAO, ImportDAO importDAO,
-                                  ObjectMapper objMapper)
+    public GermplasmImportService(Provider<GermplasmProcessor> germplasmProcessorProvider,
+                                  Provider<ProcessorManager> processorManagerProvider)
```

Comment:
You won't get a new instance unless you call .get() on a Provider for the bean. Since these processors all contain state, I want a new instance everytime process is run. If you just inject on the singleton service construction, you'll only get one instance.

Reply:
Ah, I didn't think about the GermplasmImportService being a singleton, I was just focused on the processors which are prototype. Makes sense!